bundle/endpoint/client_example_test.go

```diff
@@ -0,0 +1,89 @@
+package endpoint_test
+
+import (
+	"bytes"
+	"context"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/spiffe/go-spiffe/bundle"
+	"github.com/spiffe/go-spiffe/bundle/endpoint"
+)
+
+// This is an example of the bundle endpoint client used with the the SPIFFE
+// authentication mechanism.
+// The example assumes an SPIRE implementation of the SPIFFE bundle endpoint server
+// is running on localhost:8443 under the trust domain 'spiffe://example.org'.
+func Example_client() {
+	// Load an initial up-to-date bundle from the remote trust domain (obtained
+	// through an offline exchange).
+	rootCAs, err := loadCertificate("dummy_upstream_ca.crt")
```

Comment:
I'm not sure how useful it is to actually show how to load a certificate from disk.

Probably ok just to hand-wave this part:
```suggestion
        // TODO: load the initial set of trust domain root CAs (obtained through an offline exchange). These root CAs
        // are used to bootstrap trust with the endpoint. Afterwards, the contents of the bundle returned by the endpoint
        // should be used for future authentication.
	var rootCAs []*x509.Certificate
```